src/pages/CoreApps/CoreApps.js

```diff
@@ -0,0 +1,104 @@
+import { useDataQuery, useConfig } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { NoticeBox, CenteredContent, CircularLoader } from '@dhis2/ui'
+import React from 'react'
+import { AppsList } from '../../components/AppsList/AppsList'
+import { coreApps } from '../../core-apps'
+import { getLatestVersion } from '../../get-latest-version'
+import { semverGt } from '../../semver-gt'
+
+const query = {
+    overriddenCoreApps: {
+        resource: 'apps',
+        params: {
+            filter: 'bundled:eq:true',
+        },
+    },
+    availableCoreApps: {
+        resource: 'appHub/v2/apps',
+        params: ({ dhis_version }) => ({
+            paging: false,
+            core: true,
+            dhis_version,
+        }),
+    },
+    modules: {
+        resource: 'action::menu/getModules',
+    },
+}
+
+export const CoreApps = () => {
+    const { baseUrl, systemInfo } = useConfig()
+    const { loading, error, data } = useDataQuery(query, {
+        variables: {
+            dhis_version: systemInfo.version,
+        },
+    })
+
+    if (error) {
+        return (
+            <NoticeBox
+                error
+                title={i18n.t(
+                    'Something went wrong whilst loading your core apps'
+                )}
+            >
+                {error.message}
+            </NoticeBox>
+        )
+    }
+
+    if (loading) {
+        return (
+            <CenteredContent>
+                <CircularLoader />
+            </CenteredContent>
+        )
+    }
+
+    const appsByShortName = {}
```

Comment:
Would it be easier if we were to add a `core`-property to the API response from the App Hub? Or would we need to do this matching regardless?

Reply:
Yep adding a `core` property to the API response would eliminate the need for the `src/core-apps.js` file and the `coreApps.find(({ name }) => name === app.name)` matching.